main.go

```diff
@@ -118,35 +128,51 @@ func main() {
 		engine:         blake3Engine,
 		sliceClients:   connectToSlice(config),
 		header:         types.EmptyHeader(),
-		updateCh:      make(chan *types.Header, resultQueueSize),
+		updateCh:       make(chan *types.Header, resultQueueSize),
 		resultCh:       make(chan *types.Header, resultQueueSize),
-		previousNumber: [common.HierarchyDepth]uint64{0,0,0},
+		previousNumber: [common.HierarchyDepth]uint64{0, 0, 0},
 	}
 	log.Println("Starting Quai cpu miner in location ", config.Location)
-	m.fetchPendingHeader()
+	go m.startListener()
+	go m.fetchPendingHeader()
 	go m.subscribePendingHeader()
 	go m.resultLoop()
 	go m.miningLoop()
 	go m.hashratePrinter()
 	<-exit
 }
 
-func (m *Miner) client(ctx int) *ethclient.Client {return m.sliceClients[ctx]}
-
 // subscribePendingHeader subscribes to the head of the mining nodes in order to pass
 // the most up to date block to the miner within the manager.
-func (m *Miner) subscribePendingHeader() {
-	if _, err := m.client(common.ZONE_CTX).SubscribePendingHeader(context.Background(), m.updateCh); err != nil {
-		log.Fatal("Failed to subscribe to pending header events", err)
+func (m *Miner) subscribePendingHeader() error {
+	address := "0x4a06cbb083ce0ae76aa393db8acfced7552b7218"
+	password := "password"
+
+	msg, err := jsonrpc.MakeRequest(1, "quai_submitLogin", address, password)
```

Comment:
what is this 1 here. maybe make it a constant

Reply:
This is supposed to represent the ID for the json rpc request. Created a function to increment latestID and put that here.